frontend/fingertips-frontend/components/forms/IndicatorSearchForm/IndicatorSearchForm.test.tsx

```diff
@@ -0,0 +1,80 @@
+import { expect } from '@jest/globals';
+import { render, screen } from '@testing-library/react';
+import { IndicatorSearchFormState } from './indicatorSearchActions';
+import { IndicatorSearchForm } from '.';
+
+jest.mock('react', () => {
+  const originalModule = jest.requireActual('react');
+
+  return {
+    ...originalModule,
+    useActionState: jest
+      .fn()
+      .mockImplementation(
+        (
+          _: (
+            formState: IndicatorSearchFormState,
+            formData: FormData
+          ) => Promise<IndicatorSearchFormState>,
+          initialState: IndicatorSearchFormState
+        ) => [initialState, '/action']
+      ),
+  };
+});
+
+const initialState: IndicatorSearchFormState = {
+  indicator: '',
+  message: null,
+  errors: {},
+};
+
+it('snapshot test - renders the form', () => {
+  const container = render(
+    <IndicatorSearchForm searchFormState={initialState} />
+  );
+
+  expect(container.asFragment()).toMatchSnapshot();
+});
+
+it('should have an input field to input the indicatorId', () => {
+  render(<IndicatorSearchForm searchFormState={initialState} />);
+
+  expect(screen.getByTestId('indicator-search-form-input')).toBeInTheDocument();
```

Comment:
can you use the getByRole instead - As you have done with the test below. Getting by role is the preferred approach

Reply:
Did we want to try using getByRole instead here?